<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>trainingConfigDialog</class>
 <widget class="QDialog" name="trainingConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>624</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>624</width>
    <height>480</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>624</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="styleSheet">
   <string notr="true">#trainingConfigDialog {
	background: white;
}

QPushButton {
	font-size: 16pt;
	border: 1px solid black;
	border-radius: 10px;
	background: whitesmoke
}

#saveConfig:hover {
	color: white;
	background: green
}

#cancelConfig:hover {
	color: white;
	background: red
}

#trainingTab, #architectureTab {	
	background-color: rgb(220, 220, 220);
}

#architectureTable {
	border: none;
}

#architectureTable QHeaderView:section {
	background: whitesmoke;
}



</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="minimumSize">
        <size>
         <width>600</width>
         <height>400</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>600</width>
         <height>400</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="architectureTab">
        <attribute name="title">
         <string>Arquitectura</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout_4">
         <item row="0" column="0">
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <widget class="QLabel" name="label_11">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Muestras</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_9">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Canales</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QLabel" name="label_10">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Clases</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="timeSteps">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="value">
                <number>200</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="channels">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QSpinBox" name="classes">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="value">
                <number>4</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QGridLayout" name="gridLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="minimumSize">
                  <size>
                   <width>80</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Tipo de capa</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QDoubleSpinBox" name="dropoutRate">
                 <property name="minimumSize">
                  <size>
                   <width>60</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>60</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="cursor">
                  <cursorShape>ArrowCursor</cursorShape>
                 </property>
                 <property name="decimals">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.500000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QComboBox" name="layerType">
                 <property name="minimumSize">
                  <size>
                   <width>120</width>
                   <height>22</height>
                  </size>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <item>
                  <property name="text">
                   <string>Dense</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>LSTM</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Dropout</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="0" column="3">
                <layout class="QHBoxLayout" name="horizontalLayout_5"/>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="label_7">
                 <property name="minimumSize">
                  <size>
                   <width>60</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Unidades</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_8">
                 <property name="minimumSize">
                  <size>
                   <width>160</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Tasa (Solo para Dropout)</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="hiddenUnits">
                 <property name="minimumSize">
                  <size>
                   <width>60</width>
                   <height>22</height>
                  </size>
                 </property>
                 <property name="cursor">
                  <cursorShape>ArrowCursor</cursorShape>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>10000</number>
                 </property>
                 <property name="value">
                  <number>8</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QPushButton" name="addLayer">
                 <property name="minimumSize">
                  <size>
                   <width>30</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>30</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="icon">
                  <iconset>
                   <normalon>images/plus.png</normalon>
                  </iconset>
                 </property>
                </widget>
               </item>
               <item row="1" column="4">
                <widget class="QPushButton" name="removeLayer">
                 <property name="minimumSize">
                  <size>
                   <width>30</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>30</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="icon">
                  <iconset>
                   <normalon>images/minus.png</normalon>
                  </iconset>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QTableWidget" name="architectureTable">
               <property name="minimumSize">
                <size>
                 <width>460</width>
                 <height>240</height>
                </size>
               </property>
               <property name="tabletTracking">
                <bool>false</bool>
               </property>
               <property name="autoFillBackground">
                <bool>false</bool>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Sunken</enum>
               </property>
               <property name="sizeAdjustPolicy">
                <enum>QAbstractScrollArea::AdjustIgnored</enum>
               </property>
               <property name="selectionMode">
                <enum>QAbstractItemView::ExtendedSelection</enum>
               </property>
               <property name="textElideMode">
                <enum>Qt::ElideLeft</enum>
               </property>
               <property name="showGrid">
                <bool>true</bool>
               </property>
               <property name="gridStyle">
                <enum>Qt::DashLine</enum>
               </property>
               <property name="sortingEnabled">
                <bool>false</bool>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
               <attribute name="horizontalHeaderVisible">
                <bool>true</bool>
               </attribute>
               <attribute name="horizontalHeaderCascadingSectionResizes">
                <bool>false</bool>
               </attribute>
               <attribute name="horizontalHeaderMinimumSectionSize">
                <number>39</number>
               </attribute>
               <attribute name="horizontalHeaderDefaultSectionSize">
                <number>125</number>
               </attribute>
               <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
                <bool>false</bool>
               </attribute>
               <attribute name="horizontalHeaderStretchLastSection">
                <bool>true</bool>
               </attribute>
               <attribute name="verticalHeaderCascadingSectionResizes">
                <bool>false</bool>
               </attribute>
               <attribute name="verticalHeaderShowSortIndicator" stdset="0">
                <bool>false</bool>
               </attribute>
               <attribute name="verticalHeaderStretchLastSection">
                <bool>false</bool>
               </attribute>
               <column>
                <property name="text">
                 <string>Tipo</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Unidades</string>
                </property>
               </column>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="trainingTab">
        <attribute name="title">
         <string>Entrenamiento</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QCheckBox" name="pretrainedModelCheck">
               <property name="minimumSize">
                <size>
                 <width>220</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Entrenar modelo preentrenado</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Cargar modelo</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLineEdit" name="pretrainedModelPath">
                 <property name="minimumSize">
                  <size>
                   <width>500</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QToolButton" name="loadPretrainedModel">
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Cargra dataset</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QLineEdit" name="dataPath">
                 <property name="minimumSize">
                  <size>
                   <width>500</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QToolButton" name="loadDataPath">
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Guardar modelo</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QLineEdit" name="modelPath">
                 <property name="minimumSize">
                  <size>
                   <width>500</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QToolButton" name="saveModelPath">
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Epochs</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="epochNumber">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="cursor">
                <cursorShape>ArrowCursor</cursorShape>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_5">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Batch</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="batchSize">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="value">
                <number>32</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="saveConfig">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>40</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Guardar</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="cancelConfig">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>40</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Cancelar</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
